Archipelago.HollowKnight/Goals.cs

```diff
@@ -0,0 +1,172 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using UnityEngine.SceneManagement;
+using ItemChanger;
+
+namespace Archipelago.HollowKnight
+{
+    public enum Goals
+    {
+        Any = 0,
+        HollowKnight = 1,
+        SealedSiblings = 2,
+        Radiance = 3,
+        Godhome = 4,
+        MAX = Godhome
+    }
+
+    abstract class Goal {
+        private static readonly Dictionary<Goals, Goal> Lookup = new Dictionary<Goals, Goal>()
```

Comment:
Should probably just rename this to `GoalsLookup` or something less ambiguous in general.

Reply:
Naming is hard.  I didn't like any of the names I came up with.  That name works, so I'll use it.